arches/app/views/graph.py

```diff
@@ -678,8 +678,11 @@ def post(self, request, cardid=None):
         if self.action == "update_card":
             if data:
                 card = Card(data)
-                card.save()
-                return JSONResponse(card)
+                try:
+                    card.save()
+                    return JSONResponse(card)
+                except Exception as e:
+                    return JSONErrorResponse(content=e.args[0], status=403)
```

Comment:
I'd return 400 instead of a permssions-related error code

Reply:
Glad I asked, I didn't realize the spec had broad language about application logic. Everywhere else but the spec seems to suggest in practice 403 is only used for authorization issues, e.g [postman blog](https://blog.postman.com/what-are-http-status-codes/), [wikipedia](https://en.wikipedia.org/wiki/HTTP_403), [so](https://stackoverflow.com/questions/3290182/which-status-code-should-i-use-for-failed-validations-or-invalid-duplicates), but we can leave this as is.

(Some interesting discussion on that SO post about using 422 instead of 400 for validation, but that's too niche I think, we'd want to set up a global handler for 422's in that case if we started using it.)